src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigDocumentOptionsProviderFactory.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorLogger;
+
+namespace Microsoft.CodeAnalysis.Options.EditorConfig
+{
+    [Export(typeof(IDocumentOptionsProviderFactory)), Shared]
+    internal sealed class EditorConfigDocumentOptionsProviderFactory : IDocumentOptionsProviderFactory
```

Comment:
why is this type needed?

Reply:
also, will the changes you're making enable us to report issues with invalid editorconfig key/values?

Reply:
This type is going away, this was just being moved around as things are under construction.

Reply:
So for now since we're going to be supporting both the old and new patterns for awhile, we're going to leave this provider pattern in place since we're using that to switch between old and new. I anticipate once this is on for good, it'll perhaps get tossed.

(Background: we're merging this, but there's a number of MSBuild and project system pieces that need to be in place before this lights up. So for now, we're keeping this under a feature flag while we test things and get customer feedback. Then we'll ultimately flip the flag, and once we've fully deprecated the old one then we'll delete this.)